### ISSUE TITLE
errors when running eMouse_Shift

### ISSUE THREAD

**Issue by cchen3** [MouseLand/Kilosort2#149]
Hi, 

My computer hardware systems are: Windows 10-64bit, with MATLAB 2018a, Cuda_9.0.176, Visual Studio Community 2013 with update 5 installed; the GPU is GeForce RTX 2080. 

During KS2 installation, I have run and complete successfully mexGPUall.m in the CUDA folder. However, when I test the eMouse_Shift simulation, I got errors when running **gpuDevice(1)** and **rez = preprocessDataSub(ops)**:

![KiloSort2_eMouse Error](https://user-images.githubusercontent.com/19591595/69980602-755b1500-1528-11ea-8991-3f9ddcf1d36a.png)

Anyone have some idea on what went wrong and how to fix it please?

Many thanks!

**Comment by marius10p** [comment]
You need to get the first command "gpuDevice(1)" to work in Matlab before Kilosort2 can work at all. It is possible your GPU drivers are outdated, can you please try updating those?

**Comment by cchen3** [comment]
Thanks, Marius! I tried update the driver of GPU(GeForce RTX 2080) to the lastest (GeForce Game Ready Drive Version 441.41, Release date: 11/16/2019). But when running **gpuDevice(1);**, it got my MATLAB crash... 
![MatlabCrash_20191203](https://user-images.githubusercontent.com/19591595/70044919-20b4aa00-15bb-11ea-8e77-cd32d95891df.png)

Any idea to go from here? Thanks!

**Comment by cchen3** [comment]
Not sure if the crash report from MATLAB helps...


[Matlab Crash Report.pdf](https://github.com/MouseLand/Kilosort2/files/3916324/Matlab.Crash.Report.pdf)

**Comment by marius10p** [comment]
I don't know, this is a matlab problem, please take it up with them. You could try to reinstall Matlab. If that doesn't help, I don't know what the incompatibility is.

**Comment by dimokaramanlis** [comment]
I had this problem recently, after installing a new update for MATLAB 2019a. After updating the graphics card drivers AND reinstalling MATLAB, the problem went away...

**Comment by cchen3** [comment]
Thanks! I already reported this error to Matlab Tech Support and are now waiting for their reply. 

And I am now trying to install Matlab 2019a. Just wonder does Matlab 2019a still works with Cuda_9.0.176 and Visual Studio Community 2013, which I had already installed for Matlab 2018a? And is there any particular requirement about the order of installing these different components, such as update GPU driver -> Matlab -> cuda -> visual studio? If there is such a requirement, I may need to reinstall Cuda and Visual Studio as well?

**Comment by cchen3** [comment]
And thanks @dimokaramanlis for this very helpful information! Which graphic card do you use? Any idea to my question above please?

**Comment by marius10p** [comment]
You should probably stick with Matlab 2018a, otherwise you need a different
visual studio and CUDA.

**Comment by cchen3** [comment]
Thanks for this reminder! I will try reinstall Matlab 2018a first to see how it goes.

**Comment by cchen3** [comment]
@dimokaramanlis, I am just curious, which version of Matlab were you using when you first encountered this problem, Matlab 2019a? I am trying to think about what’s the real cause of the problem, may not be the exact version of Matlab?

**Comment by dimokaramanlis** [comment]
I was using 2019a. I noticed that the issue appeared after I installed MATLAB's update 6 for that version...

**Comment by cchen3** [comment]
I reinstalled Matlab2018a, but had the same crash... I now tend to try
MATLAB 2019b. What version of CUDA and Visual Studio do I need?

Marius Pachitariu <notifications@github.com> 于2019年12月3日周二 下午7:12写道：

> You should probably stick with Matlab 2018a, otherwise you need a different
> visual studio and CUDA.
>

**Comment by cchen3** [comment]
@dimokaramanlis, hi, what version of CUDA and Visual Studio do you use for
MATLAB 2019b please?

Dimokratis Karamanlis <notifications@github.com> 于2019年12月3日周二 下午8:38写道：

**Comment by dimokaramanlis** [comment]
CUDA toolkit 10.1 and Visual Studio 2017

**Comment by cchen3** [comment]
Thanks a lot for the reply! MATLAB Tech Support suspected there could be
something wrong with the latest driver I installed. May I ask your GPU
driver version as well?

Dimokratis Karamanlis <notifications@github.com> 于2019年12月4日周三 下午3:20写道：

**Comment by cchen3** [comment]
Hi, all, it turned out that CUDA was not properly installed on my
computer.... When I ran deviceQuery, CUDA could not detect the GPU... see
the image below. No matter whether I installed CUDA 9.0 or CUDA 10.1, it
had the same problem. I still didn't figure out why. Anyone has similar
experience or any idea please?

[image: cuda 10.1_Verification Error.PNG]

Marius Pachitariu <notifications@github.com> 于2019年12月3日周二 下午7:12写道：

> You should probably stick with Matlab 2018a, otherwise you need a different
> visual studio and CUDA.
>

**Comment by cchen3** [comment]
![cuda 10 1_Verification Error](https://user-images.githubusercontent.com/19591595/70261363-2ca29680-178a-11ea-9c40-a669d4c18a12.PNG)

**Comment by marius10p** [comment]
I would remove all Nvidia software from the computer, and re-install the latest drivers and a version of CUDA. The next step after that would be to reinstall Windows.

**Comment by cchen3** [comment]
Thanks, Marius, for these suggestions! I actually have been playing with un-installation & re-installation of driver & Cuda & Visual Studio in the last couple of days, but still did not solve the problem. I am thinking about reshape my OS to Linux. Does Kilosort2 run faster on Linux than on Windows?

**Comment by marius10p** [comment]
Not faster, similar run speeds.

**Comment by cchen3** [comment]
Hi @dimokaramanlis, did you install the professional or community version of VS 2017? Thanks!

**Comment by cchen3** [comment]
Hi all, the problem is solved on the re-installed Win 10. Thanks very much for your help and suggestions!

For reference and record if could be of any use to anyone in a similar situation:

1. GPU: Gefore RTX 2080. Install GeforeExperience and update the driver to latest Studio Driver 441.66.

2. Install the Visual Studio Community 2015. Rather than the express install as I did last time, I use customized install, to make sure all components are installed.

3. Install Cuda 10.1 with update 2. Uncheck the driver which had already been installed.

4. Build deviceQuery.exe use deviceQuery_vs2015.sln. And I then verified that CUDA could detect the GPU when ran the deviceQuery.exe, and MATLAB 2019b could detect CUDA and use VS Community 2015 complier.
